sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java

```diff
@@ -0,0 +1,243 @@
+        = "STATSBEAT_LONG_INTERVAL_SECONDS_PROPERTY_NAME";
+    private static final String STATSBEAT_SHORT_INTERVAL_SECONDS_PROPERTY_NAME
+        = "STATSBEAT_SHORT_INTERVAL_SECONDS_PROPERTY_NAME";
+
+    private static final Map<String, String> PROPERTIES
+        = CoreUtils.getProperties("azure-monitor-opentelemetry-exporter.properties");
+
+    private AzureMonitorExporterOptions exporterOptions;
+
+    private TelemetryItemExporter builtTelemetryItemExporter;
+
+    private StatsbeatModule statsbeatModule;
+
+    private ConfigProperties configProperties;
+
+    private boolean initialized;
+
+    void initializeIfNot(AzureMonitorExporterOptions exporterOptions, ConfigProperties configProperties) {
+        if (initialized) {
+            return;
+        }
+        initialized = true;
+        this.exporterOptions = exporterOptions;
+        this.configProperties = configProperties;
+        HttpPipeline httpPipeline = createHttpPipeline();
+        statsbeatModule = initStatsbeatModule(configProperties);
+        File tempDir = TempDirs.getApplicationInsightsTempDir(LOGGER,
+            "Telemetry will not be stored to disk and retried on sporadic network failures");
+        // TODO (heya) change LocalStorageStats.noop() to statsbeatModule.getNonessentialStatsbeat() when we decide to collect non-essential Statsbeat by default.
+        builtTelemetryItemExporter = AzureMonitorHelper.createTelemetryItemExporter(httpPipeline, statsbeatModule,
+            tempDir, LocalStorageStats.noop());
+        startStatsbeatModule(statsbeatModule, configProperties, tempDir); // wait till TelemetryItemExporter has been initialized before starting StatsbeatModule
+    }
+
+    SpanExporter buildSpanExporter() {
+        return new AzureMonitorTraceExporter(createSpanDataMapper(configProperties), builtTelemetryItemExporter,
+            statsbeatModule);
+    }
+
+    LogRecordExporter buildLogRecordExporter() {
+        return new AzureMonitorLogRecordExporter(
+            new LogDataMapper(true, false, createDefaultsPopulator(configProperties)), builtTelemetryItemExporter);
+    }
+
+    MetricExporter buildMetricExporter() {
+        HeartbeatExporter.start(MINUTES.toSeconds(15), createDefaultsPopulator(configProperties),
+            builtTelemetryItemExporter::send);
+        return new AzureMonitorMetricExporter(new MetricDataMapper(createDefaultsPopulator(configProperties), true),
+            builtTelemetryItemExporter);
+    }
+
+    private Set<Feature> initStatsbeatFeatures() {
+        if (System.getProperty("org.graalvm.nativeimage.imagecode") != null) {
+            return Collections.singleton(Feature.GRAAL_VM_NATIVE);
+        }
+        return Collections.emptySet();
+    }
+
+    private StatsbeatConnectionString getStatsbeatConnectionString() {
+        return StatsbeatConnectionString.create(exporterOptions.connectionString, null, null);
+    }
+
+    private SpanDataMapper createSpanDataMapper(ConfigProperties configProperties) {
```

Comment:
(and similarly in methods below since `configProperties` is now instance variable?)
```suggestion
    private SpanDataMapper createSpanDataMapper() {
```